internal/sdkv2provider/provider.go

```diff
@@ -357,7 +359,11 @@ func configure(version string, p *schema.Provider) func(context.Context, *schema
 		retryOpt := cloudflare.UsingRetryPolicy(int(retries), int(minBackOff), int(maxBackOff))
 		options := []cloudflare.Option{limitOpt, retryOpt, baseURL}
 
-		options = append(options, cloudflare.Debug(logging.IsDebugOrHigher()))
+		if logging.IsDebugOrHigher() {
```

Comment:
does this need to stay? i can't say i've used it so unsure what the goal is here.

Reply:
I'm not sure about the `cloudflare.Debug()` thing, what I'd like to add is the HTTP logging capability, it's helpful on figuring out what goes wrong by looking at the request and response payloads.

Reply:
gotcha, so the existing functionality already handles this. 

[`cloudflare.Debug()`](https://github.com/cloudflare/cloudflare-go/blob/64af3d2566da4be7093bc20562c92709628c42fa/options.go#L97) logs the HTTP transaction with some [sensitive values redacted](https://github.com/cloudflare/cloudflare-go/blob/64af3d2566da4be7093bc20562c92709628c42fa/cloudflare.go#L386-L414). the tricky bit to see here is that `logging.IsDebugOrHigher()` returns a boolean based on the setting provided to `TF_LOG`. if you want debugging enabled, you use the environment variable `TF_LOG=DEBUG terraform apply ...` to see all the logs. you can also control the [formats and flow using some other environment variables](https://developer.hashicorp.com/terraform/plugin/log/managing).

Reply:
in short, i think we can revert this and still get all the debugging goodies you are after here.

Reply:
Awesome, dropped the commit. I think I tried `TF_LOG_PROVIDER=DEBUG` and it gave nothing about the HTTP payloads, thought there is no HTTP logging in the provider. Apparently, it is related to `TF_LOG`.